docs/DeveloperGuide.md

```diff
@@ -1,15 +1,596 @@
+fast typists, it can help to process day-to-day transactions, namely your incomes and expenses. These can help you to see all your transactions
+and provide you with a valuable insight into your spending habits.
+
+This document is meant to assist potential users and developers in understanding how our program works.
+
+_Written by: Brian Wong Yun Long_
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The format of this development guide was adapted from [[SE-EDU AddressBook Level 3 Developer Guide]](https://se-education.org/addressbook-level3/DeveloperGuide.html)
+
+Some code used in this program were reused and adapted from our individual projects from the CS2113 IP. 
+
+_Written by: Brian Wong Yun Long_
+
+## Setting Up the Project
+
+{Detail how to set up the project on one's computer, assuming the software is Intellij IDEA}
+Before setting up the project on your computer, kindly check that you have installed:
+
+* Java JDK 11
+* Intellij IDEA - highly recommended
+
+Firstly, you should fork this repo, before cloning the fork to your computer.
+
+Next,
+
+1. **Ensure that Intellij JDK 11 is defined as an SDK**, as described in this [[Set up JDK guide]](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk) -- this step is not needed if you have used JDK 11 in a previous Intellij project.
+    * You _might need to set the Project language level_ section to the SDK default option.
+2. **Import the project _as a Gradle project_**, as described in
+[[se-edu's Import Gradle Project guide]](https://se-education.org/guides/tutorials/intellijImportGradleProject.html).
+3. **Running the project**: After finishing the import, locate the `src/main/java/seedu.duke/Duke.java` 
+file in this project, right-click it, and choose `Run Duke.main()`.
+
+_Written by: Paul Low_
+
+## Design
+
+### Architecture
+![Architecture Diagram](images/ArchitectureDiagram.png)
+The Architecture Diagram shown above explains the high-level design of Moolah Manager. 
+The `Duke` class contain the main method which holds the responsibility for the following:
+1. On application launch, it will initialise the `UI`, `Storage` and `Data` components. 
+2. During application execution, it will interact with `UI`, `Parser`, `Command` components 
+to execute the command entered by the users.
+3. On any exception caught, it will handle the exception and interact with the `UI` to display the error message.
+
+`Common` represents a collection of classes or enums used by multiple components.
+
+The rest of the application consists of six components:
+ - `UI`: The user interface of Moolah Manager
+ - `Parser`: Parser for user's entered command.
+ - `Command`: The command executor.
+ - `Data`: Holds the data of the application in memory.
+ - `Storage`: File I/O to store the data onto the hard disk.
```

Comment:
The software architecture section provides a clear and easily understandable description of how the software is designed.